### DISCUSSION TITLE
Layout Guidance and Pattern Contribution Process (Event Streams- Office Hours: Thu Dec 17)

### DISCUSSION THREAD

**Discussion by mindytom** [twilio-labs/paste#1053] [Office hours] [answered]
Event Streams team is working on our first UI release so that Developers can come to Event Streams to view, edit, and delete their existing sinks and subscriptions.  

Wanted to see if DSYS team have recommendations on how to lay out the Settings page (I've been referencing the [Object Details page](https://paste.twilio.design/patterns/object-details/) as well as [Transcriptions Settings Page](https://twilio.invisionapp.com/share/WXZO0CDMD49#/screens/439281148). I know DSYS don't have positions on the Editing pattern as well as command bar. Happy to understand what level of commitment is needed to help contribute a pattern or component.  

Also, for the popover component. Is there a recommendation of character count? Or type of content? 

Here's the overall [Event Streams- Release 1](https://twilio.invisionapp.com/share/HUZPA7PF6TA#/440286938_Cover_Sheet) experience. 
Here's [3 variations of the Settings page](https://twilio.invisionapp.com/share/YQZQXBEHUJP) I'm exploring
Here's how I'm thinking of using the [popover](https://twilio.invisionapp.com/share/HUZPA7PF6TA#/440266930_Index-_Popover)

**Comment by SiTaggart** [comment, marked as answer]
Hi @mindytom 

## Popover usage

![image](https://user-images.githubusercontent.com/368249/102446891-a9925e00-3fe3-11eb-90b2-28b444fc6a93.png)

Perfectly fine. There are no recommendations on length, and type of content can be anything. A Popover is a non-modal dialog, which can contain anything from something like you have, to something much more complex like an email composer you'd see in gmail. 

I'd only recommend not placing something like a table in there with hundreds of rows, otherwise it's pretty loose.

## Contribution

We've only had component contributions in the past, so that's the only thing we have to go on. Past component contributions from folks working part time on it as part of their day-to-day BU work can take up to 10 weeks in total. This includes, but not limited to, nor dictated by order:

- Kickoff
- Research, prior art, BU usage requirements gathering
- Specification writing, committee review and iteration
- Design explorations, committee review and iteration
- UX review (at least one)
- Component build
- Documentation publishing and example creation

Component build by a first time engineering contributor can take 4 weeks to complete, full time, depending on the complexity of the component.

I don't foresee a pattern contribution taking any less time if it is to be production ready. 

Alpha status components and patterns will allow for some steps to be skipped, but I'm not sure we'd accept an alpha contribution. I think it'd make most sense that a single contributor goes through all steps of creation, rather than just doing some steps and the Design Systems team taking it over to completion. Happy to be corrected by the team on that one though,

## Settings page layout

As you've rightly pointed out we don't have a position on editing just yet, and I don't think we have a position on the layout of a specific settings type of page outside of object detail patterns. This might be something best suited to a crit pod to help you iterate on and gather feedback from other parts of the app. We can then give feedback on the output of that exploration and research, and see what components might be needed if any are missing, or any specific accessibility requirements need to be considered.
